### PR TITLE
[4.0] Highlighter JS improvements

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue # .

### Summary of Changes

This PR removes the inline JS for the smart search highlighter and also removes 90% of the jQuery.

There's still a couple more lines of jQuery left to rewrite, however I wasn't sure how to test that specific function.

### Testing Instructions

1. Create a smart search module and assign it to a page.
2. Type a word in the input field and press Enter.
3. Ensure the words on the results page are highlighted

### CODE REVIEW

File: libraries/cms/html/behavior.php [hunk 1]
```diff
@@ -516,26 +516,16 @@ public static function highlighter(array $terms, $start = 'highlighter-start', $
...
-					endElement: end,
-					className: '" . $className . "',
-					onlyWords: false,
-					tag: '" . $tag . "'
-				}).highlight([\"" . implode('","', $terms) . "\"]);
-				$(start).remove();
-				$(end).remove();
-			});
-		");
+
```
Comment (Quy) [inline libraries/cms/html/behavior.php]: Remove tabs

File: media/system/js/highlighter.js [hunk 2]
```diff
@@ -1,111 +1,143 @@
...
-    return {
-        highlight: highlight,
-        unhighlight : unhighlight
-    };
-};
+Joomla = window.Joomla || {};
+
+(function(Joomla, document) {
+	'use strict';
+
```
Comment (Quy) [inline media/system/js/highlighter.js]: Remove tab

File: media/system/js/highlighter.js [hunk 3]
```diff
@@ -1,111 +1,143 @@
...
+			className: options.className,
+			onlyWords: false,
+			tag: options.tag
+		}).highlight([options.terms]);
+
+		start.parentNode.removeChild(start);
+		end.parentNode.removeChild(end);
+
+	});
+
```
Comment (Quy) [inline media/system/js/highlighter.js]: Remove tab

**Comment by dgrammatiko** [comment]
@C-Lodder please check: https://codepen.io/dgt41/pen/ZaKLbg this removes 100% of jquery and it's a custom element ;)

**Comment by C-Lodder** [comment]
1000 lines?

This could really be done in less than 100 lines. No idea why the scripts try to over complicate something so simple.

**Comment by dgrammatiko** [comment]
> 1000 lines?

I'm using an ES5 script specific for highlighting, most of the code is not mine there (not more than 100 lines, the ce part)

Why use an external script for that? Better check the possibilities in their site: https://markjs.io

**Comment by dgrammatiko** [comment]
@C-Lodder can you move this to custom element, here is something I was working on (based on markjs):https://github.com/joomla/joomla-cms/commit/1dd6004e3f1f742bbfcb59c20076f7974b8a68d8#diff-be30cdb278d8422fc9a70a5067ebf1bd

**Comment by C-Lodder** [comment]
Yeah can do. Not doing any CE stuff till your PR to remove Grunt is merged though

**Pull request closed without merging**